src/app/app.component.ts

```diff
@@ -43,4 +46,37 @@ export class AppComponent {
       }),
     );
   }
+
+  private *workPool(files: File[]) {
+    for (let file of files) {
+      yield file;
+      this.eventsPool$.next(`Main: file ${file.name} picked up for processing`);
+    }
+  }
+
+  public hashMultipleFiles(files: File[]): Observable<string> {
+    return fromWorkerPool<Blob, string>(index => {
+      const worker = new Worker('./secure-hash-algorithm.worker', { name: `sha-worker-${index}`, type: 'module' });
+      this.eventsPool$.next(`Main: worker ${index} created`);
+      return worker;
+    }, this.workPool(files)).pipe(
+      tap(res => {
+        this.eventsPool$.next(`Worker: ${res}`);
+      }),
+    );
+  }
+
+  public eventsPool$: Subject<string> = new Subject();
+  public eventListPool$: Observable<string[]> = this.eventsPool$.pipe(
+    scan<string>((list, event) => {
```

Comment:
Consider structuring the stream by filename, then projecting filtered streams in order to show the timing of each independent stream, like `+3.3sec myfile.ts : processed`

Reply:
See https://www.learnrxjs.io/operators/transformation/groupby.html